riscv/sim.cc

```diff
@@ -38,7 +38,7 @@ extern device_factory_t* ns16550_factory;
 
 sim_t::sim_t(const cfg_t *cfg, bool halted,
              std::vector<std::pair<reg_t, abstract_mem_t*>> mems,
-             std::vector<const device_factory_t*> plugin_device_factories,
+             std::vector<device_factory_t*> plugin_device_factories,
```

Comment:
It's unfortunate to have to remove `const` here. Isn't there a way you could pass the args into the factories' constructors? So we can keep `const` here?